redbeams/src/main/java/com/sequenceiq/redbeams/flow/redbeams/provision/action/RedbeamsProvisionActions.java

```diff
@@ -1,65 +1,50 @@
 package com.sequenceiq.redbeams.flow.redbeams.provision.action;
 
-import com.sequenceiq.flow.core.CommonContext;
-import com.sequenceiq.flow.core.FlowEvent;
-import com.sequenceiq.flow.core.FlowParameters;
-import com.sequenceiq.flow.core.FlowState;
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.redbeams.flow.redbeams.provision.AbstractRedbeamsProvisionAction;
+import com.sequenceiq.redbeams.flow.redbeams.provision.RedbeamsProvisionEvent;
+import com.sequenceiq.redbeams.flow.redbeams.provision.event.allocate.AllocateDatabaseServerRequest;
 import com.sequenceiq.redbeams.flow.redbeams.provision.event.allocate.AllocateDatabaseServerSuccess;
-import com.sequenceiq.redbeams.flow.stack.AbstractRedbeamsAction;
-import com.sequenceiq.redbeams.flow.stack.RedbeamsContext;
-import com.sequenceiq.redbeams.flow.stack.RedbeamsEvent;
-
-import java.util.Map;
-import java.util.Optional;
+import com.sequenceiq.redbeams.flow.redbeams.provision.RedbeamsContext;
+import com.sequenceiq.redbeams.flow.redbeams.provision.RedbeamsEvent;
+import com.sequenceiq.redbeams.flow.redbeams.provision.event.register.RegisterDatabaseServerRequest;
+import com.sequenceiq.redbeams.flow.redbeams.provision.event.register.RegisterDatabaseServerSuccess;
 
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.statemachine.StateContext;
 import org.springframework.statemachine.action.Action;
 
 @Configuration
 public class RedbeamsProvisionActions {
 
     @Bean(name = "ALLOCATE_DATABASE_STATE")
     public Action<?, ?> allocateDatabase() {
-        return new AbstractRedbeamsAction<>(RedbeamsEvent.class) {
-            @Override
-            protected RedbeamsContext createFlowContext(FlowParameters flowParameters, StateContext<FlowState, FlowEvent> stateContext,
-                                                        RedbeamsEvent payload) {
-                // Get the cloud context and the cloud credentials
-                return new RedbeamsContext(flowParameters, null, null);
-            }
+        return new AbstractRedbeamsProvisionAction<>(RedbeamsEvent.class) {
 
             @Override
-            protected void doExecute(CommonContext context, RedbeamsEvent payload, Map<Object, Object> variables) throws Exception {
-                sendEvent(context);
+            protected Selectable createRequest(RedbeamsContext context) {
+                return new AllocateDatabaseServerRequest(0L);
             }
+        };
+    }
+
+    @Bean(name = "REGISTER_DATABASE_STATE")
+    public Action<?, ?> registerDatabase() {
```

Comment:
Register is super quick because it's just writing to redbeamsdb. Maybe a flow isn't necessary? Or is it easier just to make one?

Reply:
I viewed it from the mindset of idempotency. Probably not necessary, but it can't hurt, IMO.